compile/typedef.go

```diff
@@ -43,10 +44,16 @@ func compileTypedef(file string, src *ast.Typedef) (*TypedefSpec, error) {
 		return nil, err
 	}
 
+	annotations, err := compileAnnotations(src.Annotations)
+	if err != nil {
+		return nil, err
```

Comment:
should this code builds a more complicated error? like at https://github.com/thriftrw/thriftrw-go/blob/0f849e351d9a2f5ee0553480b8e2f8b7cbc2668b/compile/enum.go#L65